src/components/Carousel/Carousel.tsx

```diff
@@ -1,176 +1,231 @@
+      setCurrentTranslate(
+        -currentIndex * (itemWidth + responsiveSlide.gap) + diff,
+      );
+    }
+  };
+
+  const handleTouchEnd = () => {
+    if (startX !== null) {
+      const distanceMoved =
+        -currentTranslate - currentIndex * (itemWidth + responsiveSlide.gap);
+
+      if (distanceMoved > 50 && currentIndex < maxIndex) {
+        setCurrentIndex(currentIndex + 1);
+      } else if (distanceMoved < -50 && currentIndex > 0) {
+        setCurrentIndex(currentIndex - 1);
       }
-    });
-  }, []); */
+
+      setAnimation(true);
+      setCurrentTranslate(-currentIndex * (itemWidth + responsiveSlide.gap));
+      setStartX(null);
+    }
+  };
+  useEffect(() => {
+    const rateComponent = document.querySelectorAll('.reviews_rate-stars');
+    const rateComponentDivs = document.querySelectorAll('.ant-rate-star div');
+    if (rateComponent) {
+      rateComponent.forEach((el) => {
+        el.setAttribute('tabindex', '-1');
+      });
+    }
+    if (rateComponentDivs) {
+      rateComponentDivs.forEach((el) => {
+        el.setAttribute('tabindex', '-1');
+      });
+    }
+  }, []);
 
   return (
-    <div
-      className={`${style.carouselContainer} ${style[classname]}`}
-      style={
-        classname === 'photos-carousel'
-          ? { backgroundColor: 'transparent' }
-          : {}
-      }
-      ref={containerRef}
-    >
-      <Carousel
-        className={sliderClassName}
-        ref={ref}
-        slidesToShow={countSlideToShow}
-        slidesToScroll={1}
-        autoplay={false}
-        dots={false}
-        initialSlide={0}
-        arrows={false}
-        responsive={
-          responsive || [
-            {
-              breakpoint: 2520,
-              settings: {
-                slidesToShow: 5,
-                slidesToScroll: 1,
-              },
-            },
-            {
-              breakpoint: 1700,
-              settings: {
-                slidesToShow: 4,
-                slidesToScroll: 1,
-              },
-            },
-            {
-              breakpoint: 1070,
-              settings: {
-                slidesToShow: 3,
-                slidesToScroll: 1,
-              },
-            },
-            {
-              breakpoint: 700,
-              settings: {
-                slidesToShow: 2,
-                slidesToScroll: 1,
-              },
-            },
-          ]
-        }
+    <div className={classNames(styles.carousel, styles[classname])}>
+      <div
+        className={styles.trackContainer}
+        onTouchStart={handleTouchStart}
+        onTouchMove={handleTouchMove}
+        onTouchEnd={handleTouchEnd}
       >
-        {children}
-      </Carousel>
-      <SliderButtons
-        handleNextClick={handleNextClick}
-        handlePrevClick={handlePrevClick}
-        isFirstSlick={isFirstSlick}
-        isLastSlick={isLastSlick}
-      />
+        <div
+          className={styles.track}
+          style={{
```

Comment:
why did you use style here? Could you use className in this case?

Reply:
here is dynamic state (currentTranslate) that's why